Framework/Geometry/src/Objects/MeshObject.cpp

```diff
@@ -0,0 +1,725 @@
+*/
+void MeshObject::setGeometryHandler(boost::shared_ptr<GeometryHandler> h) {
+  if (h == nullptr)
+    return;
+  m_handler = h;
+}
+
+/**
+* Draws the Object using geometry handler, If the handler is not set then this
+* function does nothing.
+*/
+void MeshObject::draw() const {
+  if (m_handler == nullptr)
+    return;
+  // Render the Object
+  m_handler->Render();
+}
+
+/**
+* Initializes/prepares the object to be rendered, this will generate geometry
+* for object,
+* If the handler is not set then this function does nothing.
+*/
+void MeshObject::initDraw() const {
+  if (m_handler == nullptr)
+    return;
+  // Render the Object
+  m_handler->Initialize();
+}
+
+/**
+* Returns the geometry handler
+*/
+boost::shared_ptr<GeometryHandler> MeshObject::getGeometryHandler() {
+  // Check if the geometry handler is upto dated with the cache, if not then
+  // cache it now.
+  return m_handler;
+}
+
+/**
+* Updates the geometry handler if needed
+*/
+void MeshObject::updateGeometryHandler() {
+  return; // Hopefully nothing necessary here
+}
+
+/**
+* set vtkGeometryCache writer
+*/
+void MeshObject::setVtkGeometryCacheWriter(
+    boost::shared_ptr<vtkGeometryCacheWriter> writer) {
+  m_vtk_cache_writer = writer;
+  updateGeometryHandler();
+}
+
+/**
+* set vtkGeometryCache reader
+*/
+void MeshObject::setVtkGeometryCacheReader(
+    boost::shared_ptr<vtkGeometryCacheReader> reader) {
+  m_vtk_cache_reader = reader;
+  updateGeometryHandler();
+}
+
+/**
+* Output functions for rendering, may also be used internally
+*/
+int MeshObject::numberOfTriangles() const {
+  return static_cast<int>(m_triangles.size() / 3);
+}
+
+/**
+* get faces
+*/
+int *MeshObject::getTriangles() const {
+  int *faces = nullptr;
+  size_t nFaceCorners = m_triangles.size();
+  if (nFaceCorners > 0) {
+    faces = new int[static_cast<std::size_t>(nFaceCorners)];
+    for (size_t i = 0; i < nFaceCorners; ++i) {
+      faces[i] = m_triangles[i];
+    }
+  }
+  return faces;
+}
+
+/**
+* get number of points
+*/
+int MeshObject::numberOfVertices() const {
+  return static_cast<int>(m_vertices.size());
+}
+
+/**
+* get vertices
+*/
+double *MeshObject::getVertices() const {
+  double *points = nullptr;
+  size_t nPts = m_vertices.size();
+  if (nPts > 0) {
+    points = new double[static_cast<std::size_t>(nPts) * 3];
+    for (size_t i = 0; i < nPts; ++i) {
+      V3D pnt = m_vertices[i];
+      points[i * 3 + 0] = pnt.X();
+      points[i * 3 + 1] = pnt.Y();
+      points[i * 3 + 2] = pnt.Z();
+    }
+  }
+  return points;
+}
+
+/**
+* get info on standard shapes
+*/
+void MeshObject::GetObjectGeom(int &type, std::vector<Kernel::V3D> &vectors,
+                               double &myradius, double &myheight) const {
+  type = 0;
+  if (m_handler == nullptr)
+    return;
+  m_handler->GetObjectGeom(type, vectors, myradius, myheight);
+}
+
+/** Getter for the shape xml
+@return the shape xml.
+*/
+std::string MeshObject::getShapeXML() const { return ""; }
```

Comment:
If this method can't be erased from `IObject`, the correct behaviour might be for this to throw. I think it's worth reviewing how client code is currently using this. This method does not make sense for this type.

Reply:
No code calls this.